.circleci/config.yml

```diff
@@ -641,12 +638,32 @@ jobs:
                 -DPIKA_WITH_TESTS_DEBUG_LOG=On \
                 -DPIKA_WITH_TESTS_DEBUG_LOG_DESTINATION=/build/debug-log-arm.txt \
                 -DPIKA_WITH_SPINLOCK_DEADLOCK_DETECTION=On
-          working_directory: /home/circleci/project/build
+          working_directory: $HOME/project/build
       - run:
-          name: Build pika Arm64
+          name: Building pika Arm64
           command: |
             ninja -k 0 pika
-          working_directory: /home/circleci/project/build
+          working_directory: $HOME/project/build
+      - run:
+          name: Building tests Arm64
+          command: |
+            ninja -k 0 tests.unit.modules.assertion
+          working_directory: $HOME/project/build
+      - run:
+          name: Running tests Arm64
+          when: always
+          command: |
+              ulimit -c unlimited
+              ctest -R tests.unit.modules.assertion \
+                -j4 \
+                --timeout 400 \
+                -T test \
+                --no-compress-output \
+                --output-on-failure \
+          working_directory: $HOME/project/build
+      - run:
+          <<: *convert_xml
+          working_directory: $HOME/project/build
```

Comment:
This also needs a `store_test_results` step to have the test results available for CircleCI (you can check if it's working under the "Tests" tab on the workflow page. See https://circleci.com/docs/configuration-reference#storetestresults for more details.